core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java

```diff
@@ -0,0 +1,345 @@
+				OFFICIAL_SUPPORTED_CAMEL_CATALOG_VERSIONS.add(camelVersion);
+			}
+		} catch (IOException ex) {
+			CamelModelServiceCoreActivator.pluginLog().logError("Unable to retrieve the Camel Version -> BOM Version mappings from online repo. Falling back to defaults.", ex);
+
+			// DEFAULTS
+			CAMEL_VERSION_2_FUSE_BOM_MAPPING.put("2.15.1.redhat-621084", "6.2.1.redhat-084");
+			CAMEL_VERSION_2_FUSE_BOM_MAPPING.put("2.15.1.redhat-621117", "6.2.1.redhat-117");
+			CAMEL_VERSION_2_FUSE_BOM_MAPPING.put("2.17.0.redhat-630187", "6.3.0.redhat-187");
+			CAMEL_VERSION_2_FUSE_BOM_MAPPING.put("2.17.0.redhat-630224", FUSE_63_R1_BOM_VERSION);
+			CAMEL_VERSION_2_FUSE_BOM_MAPPING.put("2.17.0.redhat-630254", FUSE_63_R2_BOM_VERSION);
+			
+			OFFICIAL_SUPPORTED_CAMEL_CATALOG_VERSIONS.add("2.17.0.redhat-630187");
+			OFFICIAL_SUPPORTED_CAMEL_CATALOG_VERSIONS.add("2.17.0.redhat-630224");
+			OFFICIAL_SUPPORTED_CAMEL_CATALOG_VERSIONS.add("2.18.1.redhat-000012");	
+			OFFICIAL_SUPPORTED_CAMEL_CATALOG_VERSIONS.add(DEFAULT_CAMEL_VERSION);
+		}
+		
+		try {
+			Properties fisMapping = new Properties();
+			URL url = new URL(FIS_MAPPING_URL);
+			fisMapping.load(url.openStream());
+		
+			for(String camelVersion : fisMapping.stringPropertyNames()) {
+				String bomVersion = fisMapping.getProperty(camelVersion);
+				PURE_FIS_CAMEL_VERSIONS.put(camelVersion, bomVersion);
+			}
+		} catch (IOException ex) {
+			CamelModelServiceCoreActivator.pluginLog().logError("Unable to retrieve the FIS-ONLY Camel Versions list from online repo. Falling back to defaults.", ex);
+
+			// DEFAULTS
+			PURE_FIS_CAMEL_VERSIONS.put("2.18.1.redhat-000012", "2.2.170.redhat-000010");			
+			PURE_FIS_CAMEL_VERSIONS.put(DEFAULT_CAMEL_VERSION,  "2.2.170.redhat-000013");
+		}
+		
+		ALL_CAMEL_CATALOG_VERSIONS.addAll(PURE_FIS_CAMEL_VERSIONS.keySet());
+		ALL_CAMEL_CATALOG_VERSIONS.addAll(OFFICIAL_SUPPORTED_CAMEL_CATALOG_VERSIONS);
+	}
+
+	private CamelCatalogUtils() {
+		// utility class
+	}
+	
+	public static List<String> getOfficialSupportedCamelCatalogVersions() {
+		return OFFICIAL_SUPPORTED_CAMEL_CATALOG_VERSIONS;
+	}
+	
+	public static List<String> getAllCamelCatalogVersions() {
+		return ALL_CAMEL_CATALOG_VERSIONS;
+	}
+	
+	/**
+	 * returns the latest and greatest supported Camel version we have a catalog 
+	 * for. If there are 2 catalogs with the same version (for instance 2.15.1 and 
+	 * 2.15.1.redhat-114) then we will always prefer the Red Hat variant.
+	 * 
+	 * @return
+	 */
+	public static String getLatestCamelVersion() {
+		return DEFAULT_CAMEL_VERSION;
+	}
+
+	/**
+	 * tries to map a FUSE BOM version to the Camel version
+	 * 
+	 * @param camelVersion
+	 * @return
+	 */
+	public static String getFuseVersionForCamelVersion(String camelVersion) {
+		return CAMEL_VERSION_2_FUSE_BOM_MAPPING.get(camelVersion);
+//		String bomVersion = CAMEL_VERSION_2_FUSE_BOM_MAPPING.get(camelVersion);
+		// TODO: revisit once https://issues.apache.org/jira/browse/CAMEL-8502 got solved
```

Comment:
What do you mean by "revisit"?
just uncommented it? or the code will need to be changed anyway?

in both conditions, what do you think to create a JIRA in Fuse Tooling tracker for that and remove the commented code?